File: fe/mobile_app/lib/models/DbModels/CommentNotification.dart
class CommentNotification 
{
  int _senderID;
  int _receiverID;
  int _postID;
  int _commentID;
  int _notificationTypeID;
  bool _read;
  DateTime _date;
  
  CommentNotification(this._senderID, this._receiverID, this._postID, this._commentID, this._notificationTypeID, this._read, this._date);

  Map<String, dynamic> toMap()
  {
    var data = Map<String, dynamic>();
    
    data["senderID"] = this._senderID;
    data["receiverID"] = this._receiverID;
    data["postID"] = this._postID;
    data["commentID"] = this._commentID;
    data["notificationTypeID"] = this._notificationTypeID;
    data["read"] = this._read;
    data["date"] = this._date.toIso8601String();

    return data;
  }
}

File: be/mytown/Models/DbModels/CommentNotification.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace mytown.Models.DbModels
{
    public class CommentNotification
    {
        public int Id { get; set; }
        public int SenderID { get; set; }
        public int ReceiverID { get; set; }
        public int PostID { get; set; }
        public int CommentID { get; set; }
        public int NotificationTypeID { get; set; }
        public bool Read { get; set; }
        public DateTime Date { get; set; }

        [ForeignKey("NotificationTypeID")]
        public NotificationType NotificationType { get; set; }

        [ForeignKey("ReceiverID")]
        public User Receiver { get; set; }

        [ForeignKey("SenderID")]
        public User Sender { get; set; }
        [ForeignKey("PostID")]
        public Post Post { get; set; }
        public virtual Comment Comment { get; set; }
    }
}
